### PR TITLE
Use container update

### PR DESCRIPTION
**Pull request by Random-Liu**
Fixes https://github.com/kubernetes-incubator/cri-containerd/issues/326.

This PR update containerd and uses the new container update function added in https://github.com/containerd/containerd/pull/1590.

### CODE REVIEW

File: vendor.conf [hunk 1]
```diff
@@ -37,7 +37,7 @@ github.com/mailru/easyjson d5b7844b561a7bc640052f1b935f7b800330d7e0
 github.com/Microsoft/go-winio v0.4.4
 github.com/opencontainers/go-digest 21dfd564fd89c944783d00d069f33e3e7123c448
 github.com/opencontainers/image-spec v1.0.0
-github.com/opencontainers/runc 593914b8bd5448a93f7c3e4902a03408b6d5c0ce
+github.com/opencontainers/runc 0351df1c5a66838d0c392b4ac4cf9450de844e2d
```
Comment (mikebrow) [inline vendor.conf]: See: https://github.com/containerd/containerd/commit/e789135aa3ffa335b6a49ec52733dc13783747a2

File: pkg/server/container_update_resources.go [hunk 2]
```diff
@@ -90,34 +93,22 @@ func (c *criContainerdService) updateContainerResources(ctx context.Context,
 				glog.Errorf("Failed to marshal spec %+v for container %q: %v", oldSpec, id, err)
 				return
 			}
-			info.Spec = any
-			if _, err := c.client.ContainerService().Update(ctx, info, "spec"); err != nil {
+			if err := cntr.Container.Update(ctx, func(ctx gocontext.Context, client *containerd.Client, c *containers.Container) error {
+				c.Spec = any
```
Comment (mikebrow) [inline pkg/server/container_update_resources.go]: Nit..  from the marshal to the call to center.Container.Update the code is duplicative with the only exception being which spec is being marshaled.. So, could be made into a update helper passing in the spec and returning one of the marshal error or the possible .Update error if Crosby's .Update is capable of generating an error outside the update function. But that could be done later...

File: pkg/server/sandbox_list.go [hunk 3]
```diff
@@ -55,7 +56,15 @@ func (c *criContainerdService) ListPodSandbox(ctx context.Context, r *runtime.Li
 			state = runtime.PodSandboxState_SANDBOX_READY
 		}
 
-		createdAt := sandboxInStore.Container.Info().CreatedAt
+		info, err := sandboxInStore.Container.Info(ctx)
+		if err != nil {
+			// It's possible that container gets deleted during list.
+			if errdefs.IsNotFound(err) {
+				continue
```
Comment (mikebrow) [inline pkg/server/sandbox_list.go]: Good!

File: vendor.conf (hunk 1 above)
Comment (Random-Liu) [reply vendor.conf]: Will update.

Comment (Random-Liu) [reply vendor.conf]: Done.

File: pkg/server/container_update_resources.go (hunk 2 above)
Comment (Random-Liu) [reply pkg/server/container_update_resources.go]: Will do.

Comment (Random-Liu) [reply pkg/server/container_update_resources.go]: Done.

File: vendor.conf (hunk 1 above)
Comment (Random-Liu) [reply vendor.conf]: We can't update now. The `docker/pkg/signal` package we are using is still using `unix.SIGUNUSED`.

However, the new `x/sys` package removed that https://github.com/containerd/containerd/commit/727fd599fba8218274a4acc3c1014823875e0e05.

Comment (Random-Liu) [reply vendor.conf]: I'll file an issue.

Comment (mikebrow) [reply vendor.conf]: yeah saw some serious changes.. thus the reason for pointing it out. To bad about the docker dependency.

**Comment by Random-Liu** [comment]
@mikebrow Addressed comments.

**Comment by Random-Liu** [comment]
Filed https://github.com/kubernetes-incubator/cri-containerd/issues/333 to track this.

File: vendor.conf (hunk 1 above)
Comment (Random-Liu) [reply vendor.conf]: Filed #333 to track this.

**Comment by mikebrow** [comment]
some odd "spurious" test result on an image pull..

**Comment by Random-Liu** [comment]
> @Random-Liu you got some odd "spurious?" test result on an image pull.. Maybe restart it..

Seems like a gcr flake. Usually happen to us in Kubernetes test. That's why we prepull image. :)

**Comment by mikebrow** [comment]
if string.Contains(image.name,"gcr.io") retry

**Pull request merged**